Automattic/jetpack
# Performance: only load concatenated CSS when needed by at least 2 modules

See #1258

## Proposed changes:

> **Note**
> This is only a partial improvement. We still need to address the larger questions around the implementation in today's world, as discussed here:
> https://github.com/Automattic/jetpack/issues/1258#issuecomment-1507814494

We currently enqueue a `jetpack.css` concatenated CSS file as soon as your site is connected to WordPress.com.
That `jetpack.css` file is built off a list of stylesheets from individual modules that are not enqueued when the concatenated file is enqueued.

In this commit, we add a new condition: for the file to be enqueued, there has to be at least 2 modules that need it. This will avoid loading the file when it wouldn't even be needed at all.
This requires a change in the implementation: we now check for active modules before we dequeue individual assets / enqueue jetpack.css. This is an additional call to get an option (see `( new Modules() )->get_active()`).
This also means that the new `$modules_with_concatenated_css` variable will need to be kept updated.

### Other information:


## Jetpack product discussion

* N/A

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

* Start with a site that's connected to WordPress.com.
* Go to Jetpack > Settings > Modules, and deactivate all modules.
* In the site's `wp-config.php` file, ensure that `SCRIPT_DEBUG` is not set to true.
* Load the site's frontend
    * You should not see any `jetpack.css` file being loaded.
* Activate the Related Posts Module.
    * There should stil be no `jetpack.css` file, but you should now see a CSS file loaded for Related Posts.
* Activate the Likes module.
    * Now you should see a `jetpack.css` file, and no longer any CSS file loaded for Related Posts or Likes.
